controllers/vrg_kubeobjects.go

```diff
@@ -157,11 +294,36 @@ func (v *VRGInstance) kubeObjectsRecover(objectStore ObjectStorer) error {
 		objectStore.AddressComponent2(),
 		v.s3KeyPrefix(),
 		// TODO query source namespace from velero backup kube object in s3 store
+		sourceNamespacedName,
 		v.instance.Namespace,
-		v.instance.Namespace,
-		VeleroNamespaceNameDefault,
+		*spec,
+		restoreIndex,
 		v.instance.Status.KubeObjectProtection.LastProtectedCapture.Number,
-	)
+	); err != nil {
+		v.log.Error(err, "kube object recover")
+		// don't return error yet since it could be non-fatal (e.g. slow API server)
+	}
+
+	// for backupName, user specifies name. for restoreName, it's generated by system.
+	restoreNamespacedName := getBackupNamespacedName(sourceNamespacedName.Name,
+		sourceNamespacedName.Namespace, fmt.Sprintf("%d", restoreIndex),
+		v.instance.Status.KubeObjectProtection.LastProtectedCapture.Number)
+
+	// can only create backup/restores in Velero/OADP namespace
+	restoreNamespacedName.Namespace = VeleroNamespaceNameDefault
+
+	// TODO: remove tight loop here
+	restoreComplete := false
+	for !restoreComplete {
+		restoreComplete, err = restoreIsDone(v.ctx, v.reconciler.APIReader,
+			objectWriter{v.reconciler.Client, v.ctx, v.log}, restoreNamespacedName)
+
+		if err != nil {
+			restoreComplete = true
+		}
+	}
```

Comment:
Forgive me for asking this question again.  Since sub-backups and sub-restores are kept around, it seems re-entry is possible and it would detect either inprogress or complete.  I'm not sure what else will be added with #25 as the objects seem to already be left.